spec/arrays.dd

```diff
@@ -1021,14 +1032,37 @@ immutable(char)[] str5 = str1.idup; // ok, make immutable copy
 
 $(SPEC_RUNNABLE_EXAMPLE_FAIL
 ---------
-char[] str1 = "abc";     // error, "abc" is not mutable
-char[] str2 = "abc".dup; // ok, make mutable copy
+char[] str1 = "abc";     // error, cannot implicitly convert expression `"abc"` of type `string` to `char[]`
+char[] str2 = "abc".dup; // ok, makes mutable copy
 string str3 = "abc";     // ok
-string str4 = str1;      // error, str4 is not mutable
-string str5 = str1.idup; // ok, make immutable copy
+string str4 = str2;      // error, cannot implicitly convert expression `str2` of type `char[]` to `string`
+string str5 = str3;      // ok, makes a mutable str5 with immutable aray contents
+string str6 = str2.idup; // ok, makes immutable copy
 ---------
 )
 
+        $(P
+        Immutable is transitive, data referred to by an immutable is
+        also immutable:
+        )
```

Comment:
This is to vague. Rephrase to: "The type `immutable(char)[]` represents an array of `immutable char`s. However, the reference to the string is mutable. If the the reference to the string needs to be `immutable` as well it can be declared `immutable char[]` or `immutable string`".

And then you can put your examples, there is no need for the second paragraph.